core/src/main/java/org/apache/accumulo/core/util/threads/Threads.java

```diff
@@ -25,6 +25,8 @@
 
 public class Threads {
 
+  private static UncaughtExceptionHandler UEH = new AccumuloUncaughtExceptionHandler();
```

Comment:
We should be taking care to avoid using JVM-wide static state. What if the user creates two clients with different exception handlers in the same JVM?

Reply:
Looking into this further, I think I want to just support a property in the client properties file and not on the AccumuloClient object. Everything in ThreadPools and Threads is static and I would need to change everything to be not-static and supply either a ClientContext or ServerContext object (which means I would have to change a ton more code). If we want to support a per-client UEH, then it might be easier to close this and update #2346 .

Reply:
I think that other PR was trying to do a lot more than what I'm suggesting here, which is just the minimal amount of work to make Threads non-static and dangling off of ClientContext, so it can be have a per-client UEH. Yes, it would be larger than this PR. Maybe it could be done as a prerequisite, or even as a follow-on, to this one. But, I think the limitation of having one client control the UEH for all clients is substantial. Other than that limitation, I really like the direction this change took.

Reply:
I was able to pull some of the ideas from #2346 into this PR. 16bc28c should address the issue that you raised here.